File: drawflow.client/src/app/features/forms/models/formTemplateVariableValidation.ts
import { Auditable } from '../../../shared/models/auditable';
import { InputValidationType } from './inputValidationType';

export interface FormTemplateVariableValidation extends Auditable {
  id: number;
  formTemplateVariableId: number;
  validationType: InputValidationType;
  validationLimit: string;
  prereqValidationType: InputValidationType | null;
  prereqValidationLimit: string | null;
  prereqValidationId: number | null;
}

File: Drawflow.Server/Data/Models/FormTemplateVariableValidation.cs
namespace Drawflow.Server.Data.Models;

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

#nullable disable

public class FormTemplateVariableValidation : Auditable
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required]
    [ForeignKey("FormTemplateVariable")]
    public long FormTemplateVariableId { get; set; }
    public virtual FormTemplateVariable FormTemplateVariable { get; set; }

    [Required]
    public InputValidationType ValidationType { get; set; }

    public string? ValidationLimit { get; set; }

    public InputValidationType? PrereqValidationType { get; set; }

    public string? PrereqValidationLimit { get; set; }

    [ForeignKey("PrereqValidation")]
    public long? PrereqValidationId { get; set; }
    public FormTemplateVariableValidation? PrereqValidation { get; set; }
}